activities2Graph.py

```diff
@@ -51,34 +51,38 @@ def idGenerator(start=0):
 
 def getEntities(projectId):
     '''get and format all entities with the inputted projectId'''
-    logging.info('Getting and formatting all entities with the inputted projectId')
-    query = syn.chunkedQuery('select * from entity where projectId =="%s"' %projectId)
-    entityDict = dict()
-    for ent in query:
-        if ent['entity.nodeType'] in [2,3,4]: 
-	#Remove containers by ignoring layers, projects, and previews
-            continue
-        for key in ent.keys():
-            #remove the "entity" portion of query
-            new_key = '.'.join(key.split('.')[1:])
-            item = ent.pop(key)
-            ent[new_key] = item[0] if (type(item) is list and len(item)>0) else item
-        ent['_type']='vertex'
-        ent['_id'] = newId.next()
-        ent['synId'] = ent.pop('id')
-
-        versionNumber = ent['versionNumber']
-        entityDict['%s.%s' %(ent['synId'],versionNumber)] = ent 
-        logging.info('Getting entity (%i): %s.%s' %(ent['_id'], ent['synId'],
-                                             ent['versionNumber']))
-        #retrieve previous versions
-        if int(versionNumber) > 1:
-            for version in range(1,int(versionNumber)):
-                ent['_id'] = newId.next()
-                entityDict['%s.%s' %(ent['synId'],version)] = ent 
-                logging.info('Getting previous version of entity (%i): %s.%i' %(ent['_id'],
-                                             ent['synId'], version))
-    return entityDict
+    global newId
+    try:
+        logging.info('Getting and formatting all entities with the inputted projectId')
+        query = syn.chunkedQuery('select * from entity where projectId =="%s"' %projectId)
+        entityDict = dict()
+        for ent in query:
+            if ent['entity.nodeType'] in [1,2,3]: 
+	    #Remove containers by ignoring layers, projects, and previews
+                continue
+            for key in ent.keys():
+                #remove the "entity" portion of query
+                new_key = '.'.join(key.split('.')[1:])
+                item = ent.pop(key)
+                ent[new_key] = item[0] if (type(item) is list and len(item)>0) else item
+            ent['_type']='vertex'
+            ent['_id'] = newId.next()
+            ent['synId'] = ent.pop('id')
+
+            versionNumber = ent['versionNumber']
+            entityDict['%s.%s' %(ent['synId'],versionNumber)] = ent 
+            logging.info('Getting entity (%i): %s.%s' %(ent['_id'], ent['synId'],
+                                                 ent['versionNumber']))
+            #retrieve previous versions
+            if int(versionNumber) > 1:
+                for version in range(1,int(versionNumber)):
```

Comment:
No guarantee that all versions between 1 and the current version exist. Use [REST call](http://hud.rel.rest.doc.sagebase.org.s3-website-us-east-1.amazonaws.com/GET/entity/id/version.html) to get all versions of a file.